components/brave_search/common/brave_search_default.mojom

```diff
@@ -0,0 +1,8 @@
+module brave_search.mojom;
+
+import "mojo/public/mojom/base/values.mojom";
+
+interface BraveSearchDefault {
+  IsDefaultSearchProvider() => (bool is_default_search_provider);
+  SetIsDefaultSearchProvider();
```

Comment:
TODO: these functions should act on the _current site_, so should accept the domain. That's so we can maintain functionality on dev / staging / production / local.

Reply:
why wouldn't you get the domain from the render frame?

Reply:
Yeah we would. What I mean here is we would pass the domain to the browser host so that the host acts on the current domain - search.brave.com dev-search.brave.com, etc.